src/windows/enumerate.rs

```diff
@@ -401,36 +401,36 @@ fn get_registry_com_ports() -> HashSet<String> {
         };
         if SUCCEEDED(query_res) {
             for idx in 0..num_key_values {
-                let mut val_name_buff = [0i8; MAX_PATH];
+                let mut val_name_buff = [0u16; MAX_PATH];
                 let mut val_name_size = MAX_PATH as u32;
                 let mut value_type = 0;
                 // if 100 chars is not enough for COM<number> something is very wrong
-                let mut val_data = [0; 100];
-                let mut data_size = val_data.len() as u32;
+                let mut val_data = [0u16; 100];
+                let mut byte_len = 2 * val_data.len() as u32; // len doubled
+
                 // SAFETY: ffi, all inputs are correct
                 let res = unsafe {
-                    RegEnumValueA(
+                    RegEnumValueW(
                         ports_key,
                         idx,
                         val_name_buff.as_mut_ptr(),
                         &mut val_name_size,
                         std::ptr::null_mut(),
                         &mut value_type,
```

Comment:
While we are here: Shouldn't we check `value_type` as well? As far as I understand the documentation of [`RegEnumValueW`](https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regenumvaluew), the value could (theoretically) be of any type and we are going to process it as wide sting data. And wouldn't [`REG_SZ`](https://learn.microsoft.com/en-us/windows/win32/sysinfo/registry-value-types) be the only variant for which we would process correctly?

Reply:
type as `REG_SZ` checks added